backend/server.py

```
@@ -46,38 +48,91 @@
-    questions = output["questions"]
-    return jsonify({"output": questions})
+    try:
+        input_text = process_input_text(input_text, use_mediawiki)
+        output = MCQGen.generate_mcq(
+            {"input_text": input_text, "max_questions": max_questions}
+        )
+        questions = output["questions"]
+        return jsonify({"output": questions})
+    except MLTimeoutError:
+        return jsonify({"error": "Request timed out. The model took too long to respond."}), 504
+    except Exception as e:
```

Ruff (0.15.7)

[warning] Do not catch blind exception: `Exception`

(BLE001)